test/support/spec_helpers/tmp.rb

```diff
@@ -73,9 +73,8 @@ def tmp(name, uniquify = true)
     #index = slash ? slash + 1 : 0
     #name.insert index, "#{SPEC_TEMP_UNIQUIFIER.succ!}-"
     slash = name.reverse.index("/")
-    index = slash ? name.size - slash - 1 : 0
+    index = slash ? name.size - slash : 0
```

Comment:
bugfix exposed when testing `Dir.chdir`, need to implement `String#rindex` and `String#insert` so that we can revert to the original ruby_mspec implementation.